src/UI/Mail/Main.hs

```diff
@@ -99,3 +108,32 @@ renderPart charsets selected hds =
         , txt pType
         ]
   in withAttr listItemAttr widget
+
+-- | render the Mailbody AST to a list used for Markup in Brick
+--
+renderMarkup ::  Maybe ScrollStep -> MailBody -> Widget Name
+renderMarkup st b =
+  let source =
+        withAttr mailbodySourceAttr $
+        padBottom (Pad 1) $ txt ("Showing output from: " <> view mbSource b)
+      bodyMarkup = toListOf (mbParagraph . to (padBottom (Pad 1) . buildParagraph st)) b
+   in source <=> vBox bodyMarkup
+
+buildParagraph :: Maybe ScrollStep -> Paragraph -> Widget Name
+buildParagraph st = vBox . toListOf (pLine . to (markup . buildWordMarkup st))
+
+-- | Render the line by inserting markup if we have a match *and* a
+-- scroll step matching
+-- Note: Why are we ignoring the line number here? Because it only
+-- matters for scrolling, not for highlighting the match.
+--
+buildWordMarkup :: Maybe ScrollStep -> Line -> Markup A.AttrName
+buildWordMarkup st (Line xs _ t) = foldr (go st) (t @? defaultAttr) xs
```

Comment:
Since the line number is only needed for scrolling, we might still be possible to get back the text wrapping using the full width of the rendering context. It needs a custom widget however.